build/files.cc

```diff
@@ -976,12 +976,12 @@ static int isHardLink(FileListRec flp, FileListRec tlp)
  */
 static int checkHardLinks(FileRecords & files)
 {
-    for (int i = 0;  i < files.size(); i++) {
+    for (rpm_count_t i = 0;  i < files.size(); i++) {
```

Comment:
I don't think rpm_count_t is right here: rpm_count_t reflects the limit inside headers, but here the data is in a vector that can hold much more, so given enough files in a buildroot, rpm_count_t could roll over and behave in various interesting ways.

I almost said it should use the same type as files.size() returns, but what this *really* should do is use iterators to avoid the issue entirely.